src/main/java/org/terasology/climateConditions/BodyTemperatureSystem.java

```diff
@@ -0,0 +1,129 @@
+            } else {
+                logger.warn("More than one entity with WorldComponent found");
+            }
+        }
+        // The world is the entity here which shall have a Body Temperature Update Periodic Action.
+    }
+
+    @ReceiveEvent
+    public void onTemperatureUpdate(PeriodicActionTriggeredEvent event, EntityRef world) {
+        if (event.getActionId().equals(BODY_TEMPERATURE_UPDATE_ACTION_ID)) {
+            for (EntityRef entity : entityManager.getEntitiesWith(AliveCharacterComponent.class)) {
+                LocationComponent location = entity.getComponent(LocationComponent.class);
+                BodyTemperatureComponent btc = entity.getComponent(BodyTemperatureComponent.class);
+                float envTemperature = climateConditionsSystem.getTemperature(location.getLocalPosition().getX(),
+                        location.getLocalPosition().getY(), location.getLocalPosition().getZ());
+                float envHumidity = climateConditionsSystem.getHumidity(location.getLocalPosition().getX(),
+                        location.getLocalPosition().getY(), location.getLocalPosition().getZ());
+                deltaTemp = ((((envTemperature - (envHumidity / 10)) - btc.bodyTemperature) / 100000) * CHECK_INTERVAL);
+                /*
+                //Send event for other systems to modify change in body temperature.
+                AffectBodyTemperatureEvent affectBodyTemperatureEvent = new AffectBodyTemperatureEvent(deltaTemp);
+                entity.send(affectBodyTemperatureEvent);
+                deltaTemp = affectBodyTemperatureEvent.getResultValue();
+                 */
+                btc.bodyTemperature = btc.bodyTemperature + deltaTemp;
+                //Check for change in body temperature levels.
+                if ((btc.bodyTemperature < 0.3) && (btc.bodyTemperature - deltaTemp > 0.3)) {
+                    entity.send(new BodyTemperatureChangedEvent(BodyTemperatureChangedEvent.BodyTemperatureLevel.LOW));
+                }
+                if ((btc.bodyTemperature > 0.3) && (btc.bodyTemperature - deltaTemp < 0.3)) {
+                    entity.send(new BodyTemperatureChangedEvent(BodyTemperatureChangedEvent.BodyTemperatureLevel.NORMAL));
+                }
+                if ((btc.bodyTemperature > 0.6) && (btc.bodyTemperature - deltaTemp < 0.6)) {
+                    entity.send(new BodyTemperatureChangedEvent(BodyTemperatureChangedEvent.BodyTemperatureLevel.LOW));
+                }
+                if ((btc.bodyTemperature < 0.6) && (btc.bodyTemperature - deltaTemp > 0.6)) {
+                    entity.send(new BodyTemperatureChangedEvent(BodyTemperatureChangedEvent.BodyTemperatureLevel.NORMAL));
+                }
+                entity.getOwner().send(new ChatMessageEvent("Body Temperature: " + btc.bodyTemperature,
+                        entity.getOwner()));
+                entity.getOwner().send(new ChatMessageEvent("Env Temperature: " + envTemperature, entity.getOwner()));
```

Comment:
Spamming the chat is most likely not the best way to indicate this to the player, but we'll find another way around this 😉